monitoring/nethealth.go

```diff
@@ -321,11 +322,16 @@ func (c *nethealthChecker) fetchNethealthMetrics(ctx context.Context) (res []byt
 		return nil, trace.Wrap(err)
 	}
 
+	req.Close = true
+
 	resp, err := client.Do(req)
+	if resp != nil {
```

Comment:
I'm not super clear about this change - according to docs, any response can be ignored if `err != nil` - did you have a specific reason for this?

Reply:
It may have gotten fixed in later releases of golang, but when researching this issue, there were a number of reports of escapes that leaked fds based on not calling resp.body.Close() even on errors from Client.Do. So IIRC some resources recommended this style of ensuring the response body is closed whenever returned. 

I don't think it's strictly necessary to address this issue, but also seems harmless to keep if it does protect against some corner case.